File: Java/app/src/main/java/com/lambdaschool/oauth2authenticationdemo/MainActivity.java
package com.lambdaschool.oauth2authenticationdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Base64;
import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        new Thread(new Runnable() {
            @Override
            public void run() {
                String auth = Base64.encodeToString("lambda-client:lambda-secret".getBytes(), Base64.DEFAULT);

                Map<String, String> headerProperties = new HashMap<>();
                headerProperties.put("Authorization", "Basic " + auth);

                /*String tokenRequest = NetworkAdapter.httpRequest(
                        "http://10.0.2.2:8080/oauth/token?grant_type=password&username=sally&password=password&scope=",
                        "POST", null, headerProperties);*/

                String tokenRequest = NetworkAdapter.httpRequest(
                        "http://10.0.2.2:8080/oauth/token?grant_type=password&username=lucy&password=ILuvM4th&scope=",
                        "POST", null, headerProperties);

                Log.i(TAG, tokenRequest);
                try {
                    String token = new JSONObject(tokenRequest).getString("access_token");

                    headerProperties.clear();
                    headerProperties.put("Authorization", "Bearer " + token);

                    String result = NetworkAdapter.httpRequest("http://10.0.2.2:8080/students/students", "GET", null, headerProperties);
                    Log.i("Authentication", result);
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }
}

File: Kotlin/app/src/main/java/com/lambdaschool/ourcoursesapp/MainActivity.kt
package com.lambdaschool.ourcoursesapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Base64
import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        MainScope().launch {
            withContext(Dispatchers.IO) {
                val auth = Base64.encodeToString("lambda-client:lambda-secret".toByteArray(), Base64.DEFAULT)

                val tokenRequest = NetworkAdapter.httpRequest(
                    "http://10.0.2.2:8080/oauth/token?grant_type=password&username=sally&password=password&scope=", "POST",
                    headerProperties = mapOf(
                        "Authorization" to "Basic $auth"
                    )
                )
                println(tokenRequest)
                val token = JSONObject(tokenRequest).getString("access_token")
                val result = NetworkAdapter.httpRequest(
                    "http://10.0.2.2:8080/students/students", "GET",
                    headerProperties = mapOf(
                        "Authorization" to "Bearer $token",
                        "Content-Type" to "application/json",
                        "Accept" to "application/json"
                    )
                )
                Log.i("Authentication", result)
            }
        }
    }
}
